### ISSUE TITLE
Memory error in calendar.TextCalendar.formatmonth

### ISSUE THREAD

**Issue by baltsers** [python/cpython#103652]
# Bug report

We did a fuzzing test on Python3.9.15, a memory error happened.

```
from calendar import *
import calendar
import argparse


def demoFunc(arg1,arg2,arg3,arg4):
    try:
        obj = calendar.TextCalendar()
        ret = obj.formatmonth(arg1,arg2,arg3,arg4)
    except (AssertionError, AttributeError, LookupError, OSError, TypeError, ValueError, argparse.ArgumentError, argparse.ArgumentTypeError, calendar.IllegalMonthError, calendar.IllegalWeekdayError) as e:
        pass

theyear=" 1106749140;00000000000000000000000000000000000000000000000000000000000000002"
themonth=11
w=11
l=11000000000000000
demoFunc(theyear,themonth,w,l)
```
[details](https://github.com/baltsers/fuzzVuls/tree/master/Python3.9.15/calendar.TextCalendar.formatmonth)

## Bug info
```
#1 0x499621 in PyMem_RawMalloc /root/CpyFuzz/experiments/Python-3.9.15/Objects/obmalloc.c:572:12
    #2 0x499621 in _PyObject_Malloc /root/CpyFuzz/experiments/Python-3.9.15/Objects/obmalloc.c:1645:11
    #3 0x4da16f in PyUnicode_New /root/CpyFuzz/experiments/Python-3.9.15/Objects/unicodeobject.c:1451:24
    #4 0x52acd7 in unicode_repeat /root/CpyFuzz/experiments/Python-3.9.15/Objects/unicodeobject.c:12917:9
    #5 0x555c48 in _PyEval_EvalFrameDefault /root/CpyFuzz/experiments/Python-3.9.15/Python/ceval.c:1621:29
    #6 0x565264 in _PyEval_EvalFrame /root/CpyFuzz/experiments/Python-3.9.15/./Include/internal/pycore_ceval.h:40:12
    #7 0x565264 in _PyEval_EvalCode /root/CpyFuzz/experiments/Python-3.9.15/Python/ceval.c:4329:14
    #8 0x43313f in _PyFunction_Vectorcall /root/CpyFuzz/experiments/Python-3.9.15/Objects/call.c:396:12
    #9 0x5635fd in _PyObject_VectorcallTstate /root/CpyFuzz/experiments/Python-3.9.15/./Include/cpython/abstract.h:118:11
    #10 0x5635fd in PyObject_Vectorcall /root/CpyFuzz/experiments/Python-3.9.15/./Include/cpython/abstract.h:127:12
    #11 0x5635fd in call_function /root/CpyFuzz/experiments/Python-3.9.15/Python/ceval.c:5077:13
    #12 0x55d897 in _PyEval_EvalFrameDefault /root/CpyFuzz/experiments/Python-3.9.15/Python/ceval.c:3506:23
    #13 0x43336f in _PyEval_EvalFrame /root/CpyFuzz/experiments/Python-3.9.15/./Include/internal/pycore_ceval.h:40:12
    #14 0x43336f in function_code_fastcall /root/CpyFuzz/experiments/Python-3.9.15/Objects/call.c:330:24
    #15 0x5635fd in _PyObject_VectorcallTstate /root/CpyFuzz/experiments/Python-3.9.15/./Include/cpython/abstract.h:118:11
    #16 0x5635fd in PyObject_Vectorcall /root/CpyFuzz/experiments/Python-3.9.15/./Include/cpython/abstract.h:127:12
    #17 0x5635fd in call_function /root/CpyFuzz/experiments/Python-3.9.15/Python/ceval.c:5077:13
    #18 0x55d96e in _PyEval_EvalFrameDefault /root/CpyFuzz/experiments/Python-3.9.15/Python/ceval.c:3520:19
    #19 0x565264 in _PyEval_EvalFrame /root/CpyFuzz/experiments/Python-3.9.15/./Include/internal/pycore_ceval.h:40:12
    #20 0x565264 in _PyEval_EvalCode /root/CpyFuzz/experiments/Python-3.9.15/Python/ceval.c:4329:14
    #21 0x554061 in _PyEval_EvalCodeWithName /root/CpyFuzz/experiments/Python-3.9.15/Python/ceval.c:4361:12
    #22 0x554061 in PyEval_EvalCodeEx /root/CpyFuzz/experiments/Python-3.9.15/Python/ceval.c:4377:12
    #23 0x554061 in PyEval_EvalCode /root/CpyFuzz/experiments/Python-3.9.15/Python/ceval.c:828:12
    #24 0x5d2367 in run_eval_code_obj /root/CpyFuzz/experiments/Python-3.9.15/Python/pythonrun.c:1221:9
    #25 0x5d2367 in run_mod /root/CpyFuzz/experiments/Python-3.9.15/Python/pythonrun.c:1242:19
    #26 0x5d2367 in pyrun_file /root/CpyFuzz/experiments/Python-3.9.15/Python/pythonrun.c:1140:15
    #27 0x5cedc6 in pyrun_simple_file /root/CpyFuzz/experiments/Python-3.9.15/Python/pythonrun.c:450:13
    #28 0x5cedc6 in PyRun_SimpleFileExFlags /root/CpyFuzz/experiments/Python-3.9.15/Python/pythonrun.c:483:15
    #29 0x41e1fd in pymain_run_file /root/CpyFuzz/experiments/Python-3.9.15/Modules/main.c:373:15
    #30 0x41e1fd in pymain_run_python /root/CpyFuzz/experiments/Python-3.9.15/Modules/main.c:598:21
    #31 0x41e1fd in Py_RunMain /root/CpyFuzz/experiments/Python-3.9.15/Modules/main.c:677:5
    #32 0x41ea66 in pymain_main /root/CpyFuzz/experiments/Python-3.9.15/Modules/main.c:707:12
    #33 0x41ead2 in Py_BytesMain /root/CpyFuzz/experiments/Python-3.9.15/Modules/main.c:731:12
    #34 0x7fe3d5768c86 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21c86)
```

# My environment

CPython: 3.9.15
Ubuntu: 18.0.4 (x86 64)

**Comment by arhadthedev** [comment]
On `Python 3.12.0a7+ (heads/main:4898415df7, Apr 20 2023, 21:40:52) [MSC v.1929 64 bit (AMD64)] on win32` I get:

```plain
Traceback (most recent call last):
  File "D:\Oleg\cpython\100.py", line 17, in <module>
    demoFunc(theyear,themonth,w,l)
  File "D:\Oleg\cpython\100.py", line 9, in demoFunc
    ret = obj.formatmonth(arg1,arg2,arg3,arg4)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Oleg\cpython\Lib\calendar.py", line 362, in formatmonth
    s += '\n' * l
         ~~~~~^~~
MemoryError
```

**Comment by arhadthedev** [comment]
The only branch that takes bug fixes is 3.11. However,  `Python 3.11.3 (tags/v3.11.3:f3909b8bc8, Apr 20 2023, 22:13:06) [MSC v.1929 64 bit (AMD64)] on win32` raises `MemoryError` too, instead of a hard crash. This behavior is expected ([`formatmonth`](https://docs.python.org/3.12/library/calendar.html?highlight=formatmonth#calendar.TextCalendar.formatmonth) cannot allocate a string with a week name taking 11,000,000,000,000,000-1 empty lines) thus I close the issue.
